lib/middleware/user-delete.js

```diff
@@ -0,0 +1,19 @@
+const { Review } = require('../models/Review');
+
+module.exports = async (req, res, next) => {
+  try {
+    const review = await Review.getById(req.params.id);
+
+    if (
+      review &&
+      (req.user.email === 'admin' || review.user_id === req.user.id)
+    ) {
+      next();
+    } else {
+      throw new Error('Access Denied!');
```

Comment:
the way this is written, if I try to delete review with id of 17 and there is no review 17, I will get the error message "Access Denied!". I think, it's hard to judge code that's not deployed by just reading it